diff --git a/muq2/CMakeLists.txt b/muq2/CMakeLists.txt
index 6f13ea2..f7824f4 100644
--- a/muq2/CMakeLists.txt
+++ b/muq2/CMakeLists.txt
@@ -17,7 +17,7 @@ IF(APPLE)
 	  execute_process(COMMAND sysctl hw.optional.arm64 OUTPUT_VARIABLE ARM_TEST OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 		# If arm64 is supported, set the CMAKE_OSX_ARCHITECTURES variable
-	  if(${ARM_TEST} STREQUAL "hw.optional.arm64: 1")
+	  if("${ARM_TEST}" STREQUAL "hw.optional.arm64: 1")
 	    message(STATUS "FOUND APPLE M1.  Setting architecture to ARM64")
 			set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "Architecture" FORCE)
 			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -arch arm64 -arch x86_64")
