diff --git a/cmake/checks/CheckBoost.cmake b/cmake/checks/CheckBoost.cmake
index 5935961..af6cbbc 100644
--- a/cmake/checks/CheckBoost.cmake
+++ b/cmake/checks/CheckBoost.cmake
@@ -4,10 +4,10 @@ set(CMAKE_REQUIRED_INCLUDES ${BOOST_INCLUDE_DIR})
 set(CMAKE_REQUIRED_FLAGS "${CMAKE_CXX_FLAGS}")
 CHECK_CXX_SOURCE_COMPILES(
 "
-#include <boost/graph/adjacency_list.hpp>
-typedef boost::adjacency_list < boost::vecS, boost::vecS, boost::bidirectionalS, int, int> Graph;
+//#include <boost/graph/adjacency_list.hpp>
+//typedef boost::adjacency_list < boost::vecS, boost::vecS, boost::bidirectionalS, int, int> Graph;
 int main(){
-Graph temp;
+//Graph temp;
 return 0;
 }
 "
@@ -19,14 +19,14 @@ CHECK_CXX_SOURCE_COMPILES(
 #include <iostream>
 #include <iterator>
 #include <algorithm>
-#include <boost/filesystem.hpp>
+//#include <boost/filesystem.hpp>
 using namespace std;
-using namespace boost::filesystem;
+//using namespace boost::filesystem;
 int main(int argc, char* argv[])
 {
-path p (argv[1]);
-directory_iterator temp(p);
-directory_iterator temp2(temp);
+//path p (argv[1]);
+//directory_iterator temp(p);
+//directory_iterator temp2(temp);
 return 0;
 }
 "
