diff --git a/NCSUOpt/DIRbatch.f b/NCSUOpt/DIRbatch.f
index a498d75b..b3ab11bf 100644
--- a/NCSUOpt/DIRbatch.f
+++ b/NCSUOpt/DIRbatch.f
@@ -12,7 +12,7 @@ C+-----------------------------------------------------------------------+
      +           dwrit,logfile,f,free,maxI,point,x,l,fmin,
      +           minpos,u,n,maxfunc,maxdeep,oops,fmax,
      +            IFeasiblef,IInfesiblef,
-     +           iidata, iisize, ddata, idsize, cdata, icsize) 
+     +           iidata, iisize, ddata, idsize) 
       IMPLICIT NONE
 
 C+-----------------------------------------------------------------------+
@@ -39,10 +39,9 @@ C+-----------------------------------------------------------------------+
 C+-----------------------------------------------------------------------+
 C| Variables to pass user defined data to the function to be optimized.  |
 C+-----------------------------------------------------------------------+
-      INTEGER iisize, idsize, icsize
+      INTEGER iisize, idsize
       INTEGER iidata(iisize)
       DOUBLE PRECISION ddata(idsize)
-      CHARACTER*40 cdata(icsize)
       
 C+-----------------------------------------------------------------------+
 C| Set the pointer to the first function to be evaluated,                |
@@ -130,17 +129,16 @@ C+-----------------------------------------------------------------------+
 C| Problem-specific Initialisation                                       |
 C+-----------------------------------------------------------------------+
       SUBROUTINE DIRInitSpecific(x,n,
-     +   iidata, iisize, ddata, idsize, cdata, icsize)
+     +   iidata, iisize, ddata, idsize)
       IMPLICIT NONE
       INTEGER n
       DOUBLE PRECISION x(n)
 C+-----------------------------------------------------------------------+
 C| Variables to pass user defined data to the function to be optimized.  |
 C+-----------------------------------------------------------------------+
-      INTEGER iisize, idsize, icsize
+      INTEGER iisize, idsize
       INTEGER iidata(iisize)
       DOUBLE PRECISION ddata(idsize)
-      CHARACTER*40 cdata(icsize)
 C+-----------------------------------------------------------------------+
 C| Problem - specific variables !                                        |
 C+-----------------------------------------------------------------------+
diff --git a/NCSUOpt/DIRect.f b/NCSUOpt/DIRect.f
index 1956aef4..372f5a24 100755
--- a/NCSUOpt/DIRect.f
+++ b/NCSUOpt/DIRect.f
@@ -44,8 +44,7 @@ C+-----------------------------------------------------------------------+
       SUBROUTINE ncsuopt_direct(fep, x, n, eps, maxf, maxT, fmin, l, u,
      +                          algmethod, Ierror, logfile, fglobal,
      +                          fglper, volper, sigmaper, iidata, 
-     +                          iisize, ddata, idsize, cdata, icsize, 
-     +                          quiet)
+     +                          iisize, ddata, idsize, quiet)
 
 C+-----------------------------------------------------------------------+
 C|    SUBROUTINE Direct                                                  |
@@ -180,10 +179,9 @@ C| User Variables.                                                       |
 C| These can be used to pass user defined data to the function to be     |
 C| optimized.                                                            |
 C+-----------------------------------------------------------------------+
-      INTEGER iisize, idsize, icsize
+      INTEGER iisize, idsize
       INTEGER iidata(iisize)
       DOUBLE PRECISION ddata(idsize)
-      Character*40 cdata(icsize)
 
       logical quiet    ! bjb
 
@@ -342,8 +340,7 @@ C+-----------------------------------------------------------------------+
       CALL DIRheader(logfile, version, x, n, eps, maxf, maxT, l, u,
      +               algmethod, maxfunc, maxdeep, fglobal, fglper,
      +               Ierror, epsfix, iepschange, volper, sigmaper,
-     +               iidata, iisize, ddata, idsize, cdata,
-     +               icsize, quiet)   ! bjb
+     +               iidata, iisize, ddata, idsize, quiet)   ! bjb
 C+-----------------------------------------------------------------------+
 C| If an error has occured while writing the header (we do some checking |
 C| of variables there), return to the main program.                      |
@@ -366,7 +363,7 @@ C+-----------------------------------------------------------------------+
 C| Start of application-specific initialisation.                         |
 C+-----------------------------------------------------------------------+
       CALL DIRInitSpecific(x,n,
-     +   iidata, iisize, ddata, idsize, cdata, icsize)
+     +   iidata, iisize, ddata, idsize)
 C+-----------------------------------------------------------------------+
 C| End of application-specific initialisation.                           |
 C+-----------------------------------------------------------------------+
@@ -391,7 +388,7 @@ C+-----------------------------------------------------------------------+
       CALL DIRpreprc(u,l,n,l,u,oops)
       IF (oops .GT. 0) THEN
         Write(*,10005)
-        Write(logfile,10005)
+        IF (logfile .gt. 0) Write(logfile,10005)
         IError = -3
         Return
       END IF
@@ -406,19 +403,19 @@ C+-----------------------------------------------------------------------+
      +   dwrit,logfile,ArrayI,maxI,List2,w,x,l,u,fmin,minpos,
      +   thirds,levels,maxfunc,maxdeep,n,MaxDim,fmax,Ifeasiblef,
      +   IInfesiblef, Ierror,
-     +   iidata, iisize, ddata, idsize, cdata, icsize)
+     +   iidata, iisize, ddata, idsize)
 C+-----------------------------------------------------------------------+
 C| Added error checking.                                                 |
 C+-----------------------------------------------------------------------+
       IF (Ierror .lt. 0) then
          IF (Ierror .eq. -4) THEN
             Write(*,10006)
-            Write(logfile,10006)
+            IF (logfile .gt. 0) Write(logfile,10006)
             return
          END IF
          IF (Ierror .eq. -5) THEN
             Write(*,10007)
-            Write(logfile,10007)
+            IF (logfile .gt. 0) Write(logfile,10007)
             return
          END IF
       END IF
@@ -434,10 +431,10 @@ C| the iteration, the number of function evaluations done and fmin.      |
 C+-----------------------------------------------------------------------+
       IF (Ifeasiblef .gt. 0) then
         if (.not. quiet) write(*,10012) tstart-1,numfunc     ! bjb
-        write(logfile,10012) t,numfunc
+        IF (logfile .gt. 0) write(logfile,10012) tstart-1,numfunc
       ELSE
         if (.not. quiet) Write(*,10002) numfunc, fmin, fmax     ! bjb
-        Write(logfile,10003) tstart-1,numfunc,fmin
+        IF (logfile .gt. 0) Write(logfile,10003) tstart-1,numfunc,fmin
       END IF
 C+-----------------------------------------------------------------------+
 C+-----------------------------------------------------------------------+
@@ -467,10 +464,12 @@ C+-----------------------------------------------------------------------+
               Write(*,10021)
               Write(*,10022)
               Write(*,10023)
-              Write(logfile,10020)
-              Write(logfile,10021)
-              Write(logfile,10022)
-              Write(logfile,10023)
+              IF (logfile .gt. 0) THEN
+                 Write(logfile,10020)
+                 Write(logfile,10021)
+                 Write(logfile,10022)
+                 Write(logfile,10023)
+              END IF
               return
           END IF
         ENDIF
@@ -497,7 +496,7 @@ C| dept, stop the computation.                                           |
 C+-----------------------------------------------------------------------+
               IF (actdeep+1 .GE. mdeep) THEN
                  Write(*,10004)
-                 write(logfile,10004)
+                 IF (logfile .gt. 0) write(logfile,10004)
                  Ierror = -6
                  GOTO 100
               END IF
@@ -530,13 +529,13 @@ C+-----------------------------------------------------------------------+
      +             x,l,fmin,minpos,u,n,maxfunc,maxdeep,oops)
               IF (oops .GT. 0) THEN
                 Write(*,10006)
-                Write(logfile,10006)
+                IF (logfile .gt. 0) Write(logfile,10006)
                 IError = -4
                 return
               END IF
               Newtosample = newtosample + maxI
               CALL fep(n,c,l,u,point,maxI,start,maxfunc,f,
-     +                 iidata, iisize, ddata, idsize, cdata, icsize)
+     +                 iidata, iisize, ddata, idsize)
 C+-----------------------------------------------------------------------+
 C| JG 01/22/01 Added variable to keep track of the maximum value found.  |
 C+-----------------------------------------------------------------------+
@@ -544,10 +543,10 @@ C+-----------------------------------------------------------------------+
      +            length,dwrit,logfile,f,free,maxI,point,x,l,
      +            fmin,minpos,u,n,maxfunc,maxdeep,oops,fmax,
      +            Ifeasiblef,IInfesiblef,
-     +            iidata, iisize, ddata, idsize, cdata, icsize)
+     +            iidata, iisize, ddata, idsize)
               IF (oops .GT. 0) THEN
                 Write(*,10007)
-                Write(logfile,10007)
+                IF (logfile .gt. 0) Write(logfile,10007)
                 IError = -5
                 return
               END IF
@@ -577,7 +576,7 @@ C| in the array.                                                         |
 C+-----------------------------------------------------------------------+
         IF (oldpos .LT. minpos) THEN
           if (.not. quiet) Write(*,10002) numfunc,fmin, fmax     ! bjb
-          Write(logfile,10003) t,numfunc,fmin
+          IF (logfile .gt. 0) Write(logfile,10003) t,numfunc,fmin
         END IF
 C+-----------------------------------------------------------------------+
 C| If no feasible point has been found, give out the iteration, the      |
@@ -585,7 +584,7 @@ C| number of function evaluations and a warning.                         |
 C+-----------------------------------------------------------------------+
         IF (Ifeasiblef .gt. 0) then
           if (.not. quiet) write(*,10012) t,numfunc     ! bjb
-          write(logfile,10012) t,numfunc
+          IF (logfile .gt. 0) write(logfile,10012) t,numfunc
         END IF
 C+-----------------------------------------------------------------------+
 C+-----------------------------------------------------------------------+
@@ -612,7 +611,7 @@ C+-----------------------------------------------------------------------+
         IF (delta .LE. volper) THEN
            Ierror = 4
            if (.not. quiet) Write(*,10011) delta, volper    ! bjb
-           Write(logfile,10011) delta, volper
+           IF (logfile .gt. 0) Write(logfile,10011) delta, volper
            GOTO 100
         END IF
 C+-----------------------------------------------------------------------+
@@ -625,7 +624,7 @@ C+-----------------------------------------------------------------------+
         IF (delta .LE. sigmaper) THEN
            Ierror = 5
            if (.not. quiet) Write(*,10013) delta, sigmaper     ! bjb
-           Write(logfile,10013) delta, sigmaper
+           IF (logfile .gt. 0) Write(logfile,10013) delta, sigmaper
            GOTO 100
         END IF
 C+-----------------------------------------------------------------------+
@@ -637,7 +636,7 @@ C+-----------------------------------------------------------------------+
      +   THEN
            Ierror = 3
            if (.not. quiet) Write(*,10010)     ! bjb
-           Write(logfile,10010)
+           IF (logfile .gt. 0) Write(logfile,10010)
            GOTO 100
         END IF
 C+-----------------------------------------------------------------------+
@@ -668,7 +667,7 @@ C+-----------------------------------------------------------------------+
         IF (increase .eq. 1) then        
            maxf = numfunc + oldmaxf
            IF (Ifeasiblef .eq. 0) then
-             write(logfile,10031) maxf
+             IF (logfile .gt. 0) write(logfile,10031) maxf
              increase = 0
            END IF
         END IF
@@ -682,11 +681,11 @@ C+-----------------------------------------------------------------------+
            IF (Ifeasiblef .eq. 0) then
               Ierror = 1
               if (.not. quiet) Write(*,10008)     ! bjb
-              Write(logfile,10008)
+              IF (logfile .gt. 0) Write(logfile,10008)
               GOTO 100
            ELSE
               increase = 1
-              write(logfile,10030) numfunc
+              IF (logfile .gt. 0) write(logfile,10030) numfunc
               maxf = numfunc+ oldmaxf
            END IF
         END IF
@@ -702,7 +701,7 @@ C| The algorithm stopped after maxT iterations.                          |
 C+-----------------------------------------------------------------------+
       Ierror = 2
       if (.not. quiet) Write(*,10009)     ! bjb
-      Write(logfile,10009)
+      IF (logfile .gt. 0) Write(logfile,10009)
 
 100   CONTINUE
 C+-----------------------------------------------------------------------+
diff --git a/NCSUOpt/DIRsubrout.f b/NCSUOpt/DIRsubrout.f
index 0667b77e..e9f354bb 100755
--- a/NCSUOpt/DIRsubrout.f
+++ b/NCSUOpt/DIRsubrout.f
@@ -82,15 +82,17 @@ C+-----------------------------------------------------------------------+
                help2 = thirds(S(i,2)) - thirds(S(j,2))
                help2 = (f(i_,1) - f(j_,1))/help2
                IF (help2 .LE. 0.D0) THEN
-                 IF (dwrit .EQ. 2) THEN
-                   Write(logfile,*) "thirds > 0,help2 <= 0"
-                 END IF
+                IF (dwrit .EQ. 2) THEN
+                   IF (logfile .gt. 0)
+     +                Write(logfile,*) "thirds > 0,help2 <= 0"
+                END IF
                  GOTO 60
                END IF
                IF (help2 .LT. helplower) THEN
-                 IF (dwrit .EQ. 2) THEN
-                   Write(logfile,*) "helplower = ",help2
-                 END IF
+                IF (dwrit .EQ. 2) THEN
+                   IF (logfile .gt. 0)
+     +                Write(logfile,*) "helplower = ",help2
+                END IF
                  helplower = help2
                END IF
          	   END IF
@@ -109,15 +111,17 @@ C+-----------------------------------------------------------------------+
                help2 = thirds(S(i,2)) - thirds(S(j,2))
                help2 = (f(i_,1) - f(j_,1))/help2
                IF (help2 .LE. 0.D0) THEN
-                 IF (dwrit .EQ. 2) THEN
-                   Write(logfile,*) "thirds < 0,help2 <= 0"
-                 END IF
+                IF (dwrit .EQ. 2) THEN
+                   IF (logfile .gt. 0)
+     +                Write(logfile,*) "thirds < 0,help2 <= 0"
+                END IF
                  GOTO 60
                END IF
                IF (help2 .GT. helpgreater) THEN
-                 IF (dwrit .EQ. 2) THEN
-                   Write(logfile,*) "helpgreater = ",help2
-                 END IF
+                IF (dwrit .EQ. 2) THEN
+                   IF (logfile .gt. 0)
+     +                Write(logfile,*) "helpgreater = ",help2
+                END IF
                  helpgreater = help2
                END IF
              END IF
@@ -135,14 +139,16 @@ C+-----------------------------------------------------------------------+
            IF ((f(j_,1) - helplower * thirds(S(j,2))) .GT. 
      +        (fmin - eps*abs(fmin))) THEN
               IF (dwrit .EQ. 2) THEN
-                Write(logfile,*) "> fmin - eps|fmin|"
+                IF (logfile .gt. 0)
+     +             Write(logfile,*) "> fmin - eps|fmin|"
               END IF
               GOTO 60
             END IF
          ELSE
            IF (dwrit .EQ. 2) THEN
-           Write(logfile,*) "helpgreater > helplower",helpgreater,
-     +            helplower,helpgreater - helplower
+              IF (logfile .gt. 0)
+     +           Write(logfile,*) "helpgreater > helplower",
+     +              helpgreater, helplower, helpgreater - helplower
            END IF
            GOTO 60
          END IF
@@ -445,8 +451,9 @@ C+-----------------------------------------------------------------------+
                pos = point(pos)
             end if
             if (pos .eq. 0) then
-               write(logfile,*) 'Error in DIRREsortlist: We went ',
-     + 'through the whole list and could not find the point to 
+               IF (logfile .gt. 0)
+     +            write(logfile,*) 'Error in DIRREsortlist: We went ',
+     +            'through the whole list and could not find the point to 
      +  replace!!'
                goto 20
             end if
@@ -652,7 +659,7 @@ C+-----------------------------------------------------------------------+
      + dwrit,logfile,ArrayI,maxI,List2,w,x,l,u,fmin,minpos,thirds,
      + levels,maxfunc,maxdeep,n,maxor,fmax,Ifeasiblef,IInfeasible, 
      + Ierror,
-     + iidata, iisize, ddata, idsize, cdata, icsize)
+     + iidata, iisize, ddata, idsize)
       IMPLICIT None
       Integer maxfunc,maxdeep,n,maxor
       Double Precision  f(maxfunc,2),c(maxfunc,maxor),fmin
@@ -683,10 +690,9 @@ C JG 09/15/00 Added variable JONES (see above)
 C+-----------------------------------------------------------------------+
 C| Variables to pass user defined data to the function to be optimized.  |
 C+-----------------------------------------------------------------------+
-      INTEGER iisize, idsize, icsize
+      INTEGER iisize, idsize
       INTEGER iidata(iisize)
       Double Precision ddata(idsize)
-      Character*40 cdata(icsize)
         
       fmin = 1.D20
       costmin = fmin
@@ -724,7 +730,7 @@ C JG 09/15/00 Initialiase levels to contain 1/j
         length(1,i) = 0
 20    CONTINUE
       CALL fep(n,x,l,u,0,0,1,maxfunc,f,
-     +         iidata, iisize, ddata, idsize, cdata, icsize)
+     +         iidata, iisize, ddata, idsize)
       help=f(1,2)
 C+------------------------------------------------------------------+
 C|      CALL DIRinfcn(fcn,x,l,u,n,f(1,1),help,
@@ -766,7 +772,7 @@ C+-----------------------------------------------------------------------+
          return
       END IF
       CALL fep(n,c,l,u,point,maxI,new,maxfunc,f,
-     +         iidata, iisize, ddata, idsize, cdata, icsize)
+     +         iidata, iisize, ddata, idsize)
 C+-----------------------------------------------------------------------+
 C| JG 01/22/01 Added variable to keep track of the maximum value found.  |
 C|             Added variable to keep track if feasible point was found. |
@@ -775,7 +781,7 @@ C+-----------------------------------------------------------------------+
      +         dwrit,logfile,f,free,maxI,point,x,l,
      +         fmin,minpos,u,n,maxfunc,maxdeep,oops,fmax,Ifeasiblef,
      +         IInfeasible,
-     +         iidata, iisize, ddata, idsize, cdata, icsize)
+     +         iidata, iisize, ddata, idsize)
 C+-----------------------------------------------------------------------+
 C| JG 01/23/01 Added error checking.                                     |
 C+-----------------------------------------------------------------------+
@@ -870,8 +876,10 @@ C+-----------------------------------------------------------------------+
            Write(*,1000) 
            Write(*,1001) 
            IF (dwrit .EQ. 2) THEN
-             Write(logfile,1000)
-             Write(logfile,1001)
+             IF (logfile .gt. 0) THEN
+                Write(logfile,1000)
+                Write(logfile,1001)
+             END IF
            END IF
            oops = 1
            RETURN
@@ -890,7 +898,7 @@ C+-----------------------------------------------------------------------+
       IF (pos .GT. 0) THEN
           Write(*,2000)
           IF (dwrit .EQ. 2) THEN
-             Write(logfile,2000)
+             IF (logfile .gt. 0) Write(logfile,2000)
            END IF
           STOP
 2000      FORMAT("Error ! ") 
@@ -1184,7 +1192,7 @@ C| The subroutine whose name is passed through the argument fcn.         |
 C|                                                                       |
 C+-----------------------------------------------------------------------+
       subroutine DIRinfcn(fcn,x,c1,c2,n,f,flag,
-     +                  iidata, iisize, ddata, idsize, cdata, icsize)
+     +                  iidata, iisize, ddata, idsize)
 
       implicit none
       integer n,i, flag
@@ -1194,10 +1202,9 @@ C+-----------------------------------------------------------------------+
 C+-----------------------------------------------------------------------+
 C| Variables to pass user defined data to the function to be optimized.  |
 C+-----------------------------------------------------------------------+
-      INTEGER iisize, idsize, icsize
+      INTEGER iisize, idsize
       INTEGER iidata(iisize)
       Double Precision ddata(idsize)
-      Character*40 cdata(icsize)
 
 
 C+-----------------------------------------------------------------------+
@@ -1210,7 +1217,7 @@ C+-----------------------------------------------------------------------+
 C| Call the function-evaluation subroutine fcn.                          |
 C+-----------------------------------------------------------------------+
       f = 0.D0
-      CALL fcn(n,x,f,flag,iidata, iisize, ddata, idsize, cdata, icsize)
+      CALL fcn(n,x,f,flag,iidata, iisize, ddata, idsize)
 
 C+-----------------------------------------------------------------------+
 C| Rescale the variable x.                                               |
@@ -1260,8 +1267,7 @@ C+-----------------------------------------------------------------------+
      +               algmethod, maxfunc, maxdeep,    
      +               fglobal, fglper, Ierror,epsfix, iepschange,
      +               volper, sigmaper, 
-     +               iidata, iisize, ddata, idsize, cdata, icsize, 
-     +               quiet)
+     +               iidata, iisize, ddata, idsize, quiet)
       IMPLICIT None
       Integer logfile, version,n, maxf, maxT
       Integer algmethod, Ierror, i, maxfunc, maxdeep
@@ -1272,16 +1278,15 @@ C+-----------------------------------------------------------------------+
 C+-----------------------------------------------------------------------+
 C| Variables to pass user defined data to the function to be optimized.  |
 C+-----------------------------------------------------------------------+
-      INTEGER iisize, idsize, icsize
+      INTEGER iisize, idsize
       INTEGER iidata(iisize)
       Double Precision ddata(idsize)
-      Character*40 cdata(icsize)
       
       Integer Imainver, Isubver, Isubsubver, Ihelp, numerrors
 
       logical quiet     ! bjb
 
-      Write(logfile,900)
+      IF (logfile .gt. 0) Write(logfile,900)
       numerrors = 0
       IError = 0
       Imainver = INT(version/100)
@@ -1304,7 +1309,10 @@ C+-----------------------------------------------------------------------+
         epsfix = 1.D100
       endif
       
-      write(logfile,100) Imainver, Isubver, Isubsubver
+      IF (logfile .gt. 0) then
+         write(logfile,100) Imainver, Isubver,
+     +      Isubsubver
+      end if
       if (.not. quiet) then   ! bjb
          write(*,100) Imainver, Isubver, Isubsubver
 C+-----------------------------------------------------------------------+
@@ -1329,31 +1337,35 @@ C+-----------------------------------------------------------------------+
 C| JG 07/16/01 Removed printout of contents in cdata(1).                 |
 C+-----------------------------------------------------------------------+
 C      write(logfile,*) cdata(1)
-      write(logfile,200) n
-      write(logfile,201) eps
-      write(logfile,202) maxf
-      write(logfile,203) maxT
-      write(logfile,204) fglobal
-      write(logfile,205) fglper
-      write(logfile,208) volper
-      write(logfile,209) sigmaper
-      if (iepschange .eq. 1) then
-         write(logfile,206)
-      else
-         write(logfile,207)
-      end if
+      IF (logfile .gt. 0) THEN
+         write(logfile,200) n
+         write(logfile,201) eps
+         write(logfile,202) maxf
+         write(logfile,203) maxT
+         write(logfile,204) fglobal
+         write(logfile,205) fglper
+         write(logfile,208) volper
+         write(logfile,209) sigmaper
+         if (iepschange .eq. 1) then
+            write(logfile,206)
+         else
+            write(logfile,207)
+         end if
+      END IF
       if (algmethod .eq. 0) then
          if (.not. quiet) then     ! bjb
             write(*,*) 'Jones original DIRECT algorithm is used.'
          end if
-         write(logfile,*) 'Jones original DIRECT algorithm is used.'
+         IF (logfile .gt. 0)
+     +      write(logfile,*) 'Jones original DIRECT algorithm is used.'
       else
          if (.not. quiet) then     ! bjb
             write(*,*) 'Our modification of the DIRECT algorithm',
      +                 ' is used.'
          end if
-         write(logfile,*) 'Our modification of the DIRECT algorithm',
-     +                    ' is used.'
+         IF (logfile .gt. 0)
+     +      write(logfile,*) 'Our modification of the DIRECT algorithm',
+     +                         ' is used.'
       end if
       do 1010, i = 1,n
          IF (u(i) .le. l(i)) then
@@ -1361,13 +1373,13 @@ C      write(logfile,*) cdata(1)
             if (.not. quiet) then     ! bjb
                write(*,153) i,l(i), u(i)
             end if
-            write(logfile,153) i,l(i), u(i)
+            IF (logfile .gt. 0) write(logfile,153) i,l(i), u(i)
             numerrors = numerrors + 1
          else
             if (.not. quiet) then     ! bjb
                write(*,152) i,l(i), u(i)
             end if
-            write(logfile,152) i,l(i), u(i)
+            IF (logfile .gt. 0) write(logfile,152) i,l(i), u(i)
          end if
 1010  continue
 C+-----------------------------------------------------------------------+
@@ -1377,25 +1389,26 @@ C| occurred, we give out an extra message.                               |
 C+-----------------------------------------------------------------------+
       IF ((maxf+20) .GT. maxfunc) THEN
          Write(*,10001) maxf, maxfunc
-         Write(logfile,10001) maxf, maxfunc
+         IF (logfile .gt. 0) Write(logfile,10001) maxf, maxfunc
          numerrors = numerrors + 1
          IError = -2
       END IF
       if (IError .lt. 0) then
-         write(logfile,120)
+         IF (logfile .gt. 0) write(logfile,120)
          if (.not. quiet) write(*,120)     ! bjb
          if (numerrors .eq. 1) then
             write(*,105)
-            write(logfile,105)
+            IF (logfile .gt. 0) write(logfile,105)
          else
             write(*,110) numerrors
-            write(logfile,110) numerrors
+            IF (logfile .gt. 0) write(logfile,110) numerrors
          end if
       end if
-      write(logfile,120)
+      IF (logfile .gt. 0) write(logfile,120)
       if (.not. quiet) write(*,120)     ! bjb
       if (IError .ge. 0) then
-         write(logfile,*) 'Iteration   # of f-eval.   fmin'
+         IF (logfile .gt. 0)
+     +      write(logfile,*) 'Iteration   # of f-eval.   fmin'
       end if
 
 10001 FORMAT("WARNING : The maximum number of function evaluations (",
@@ -1438,17 +1451,20 @@ C+-----------------------------------------------------------------------+
       Double Precision x(n), l(n), u(n)
       Double Precision fglobal , fmin
 
-      Write(logfile,900)
-      Write(logfile,1000) fmin
-      Write(logfile,1010) numfunc
-      if (fglobal .gt. -1.D99) then
-         write(logfile,1001) 100*(fmin-fglobal)/max(1.D0,abs(fglobal))
-      end if
-      Write(logfile,1002) 
-      do 100, i = 1,n
-         write(logfile,1003) i, x(i), x(i)-l(i), u(i) - x(i)
-100   continue
-      write(logfile,1200)
+      IF (logfile .gt. 0) THEN
+         Write(logfile,900)
+         Write(logfile,1000) fmin
+         Write(logfile,1010) numfunc
+         if (fglobal .gt. -1.D99) then
+            write(logfile,1001)
+     +         100*(fmin-fglobal)/max(1.D0,abs(fglobal))
+         end if
+         Write(logfile,1002) 
+         do 100, i = 1,n
+            write(logfile,1003) i, x(i), x(i)-l(i), u(i) - x(i)
+100      continue
+         write(logfile,1200)
+      END IF
       
 900   FORMAT('--------------------------------- Summary -------------'
      +       ,'-------------------')
@@ -1473,10 +1489,12 @@ C+-----------------------------------------------------------------------+
       Write(*,10003)
       Write(*,10004)
       IF (dwrit .EQ. 2) THEN
-         Write(logfile,10001) maxf
-         Write(logfile,10002) maxfunc
-         Write(logfile,10003)
-         Write(logfile,10004)
+         IF (logfile .gt. 0) THEN
+            Write(logfile,10001) maxf
+            Write(logfile,10002) maxfunc
+            Write(logfile,10003)
+            Write(logfile,10004)
+         END IF
       END IF
 
 10001 FORMAT("The maximum number of function evaluations (",
@@ -1495,9 +1513,11 @@ C+-----------------------------------------------------------------------+
       Write(*,10002) maxdeep
       Write(*,10003)
       IF (dwrit .EQ. 2) THEN
-         Write(logfile,10001) maxT
-         Write(logfile,10002) maxdeep
-         Write(logfile,10003)
+         IF (logfile .gt. 0) THEN
+            Write(logfile,10001) maxT
+            Write(logfile,10002) maxdeep
+            Write(logfile,10003)
+         END IF
       END IF
 
 10001 FORMAT("The maximum number of iterations (",I5,
